File: LeetCode/Easy/0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.java
class Solution {
    public int[] shortestToChar(String s, char c) {
        char[] sCharArray = s.toCharArray();
        int n = sCharArray.length;
        int[] ret = new int[n];
        int val = 9999;
        
        for (int i = 0; i < n; ++i){
            if (sCharArray[i] == c){
                ret[i] = 0;
                val = 1;
            } else {
                ret[i] = val++;
            }
        }
        
        val = 9999;
        
        for (int i = n - 1; i > -1; i--){
            if (sCharArray[i] == c){
                ret[i] = 0;
                val = 1;
            } else {
                ret[i] = Math.min(ret[i], val++);
            }
        }
        
        return ret;
    }
}

File: LeetCode/Easy/0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
      n = len(s)
      ret = [n] * n

      val = n
      
      for i in range(n):
        if s[i] == c:
          ret[i] = 0
          val = 1
        else:
          ret[i] = val
          val += 1
      
      val = n

      for i in range(n - 1, -1, -1):
        if s[i] == c:
          ret[i] = 0
          val = 1
        else:
          ret[i] = min(val, ret[i])
          val += 1

      return ret